cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/AppLifecycleHttpHandlerTest.java

```diff
@@ -136,9 +138,9 @@ public void testDeployVersionedAndNonVersionedApp() throws Exception {
     ConfigTestApp.ConfigClass config = new ConfigTestApp.ConfigClass("abc", "def");
     AppRequest<ConfigTestApp.ConfigClass> request = new AppRequest<>(
       new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config);
-    Assert.assertEquals(200, deploy(appId, request).getStatusLine().getStatusCode());
+    Assert.assertEquals(200, deployVersion(appId, request).getStatusLine().getStatusCode());
```

Comment:
but is it treated differently? does version completely change the logic of the method? the contract of the method (to deploy an app) should not change whether the app has a version or not, right?